sparkly/utils.py

```diff
@@ -183,3 +182,102 @@ def _init_array(*args):
     'struct': _init_struct,
     'array': _init_array,
 }
+
+
+class KafkaWatcher():
+    """Context manager that tracks Kafka data published to a topic
+
+    Provides access to the new items that were written to a kafka topic by code running
+    within this context.  This is mainly useful in integration test cases.
+
+    Usage:
+        my_deserializer = lambda item: json.loads(item.decode('utf-8'))
+        kafka_watcher = KafkaWatcher(
+            my_sparkly_session,
+            expected_output_dataframe_schema,
+            my_deserializer,
+            my_deserializer,
+            'my.kafkaserver.net',
+            'my_kafka_topic',
+        )
+        with kafka_watcher:
+            # do stuff that publishes messages to 'my_kafka_topic'
+        self.assertEqual(kafka_watcher.count, expected_number_of_new_messages)
+        self.assertDataFrameEqual(kafka_watcher.df, expected_df)
+    """
+
+    count = 0
+    _df = None
```

Comment:
I think you can already nest it, because once we set it with `self.count = ...`  in `__enter__()` you end up setting a new attribute on the instance instead of modifying the class attribute.  I moved it anyway, to make it clearer.